lib/libzrepl/data_conn.c

```diff
@@ -434,6 +436,9 @@ uzfs_zvol_worker(void *arg)
 	(void) pthread_mutex_unlock(&zinfo->zinfo_mutex);
 
 drop_refcount:
+	atomic_add_64(&zinfo->inflight_io_cnt, -1);
+	atomic_add_64(&zinfo->dispatched_io_cnt, -1);
```

Comment:
Apologies for the late comment, but in general we prefer to only have incrementing counters. If we need to know the difference between inflight and dispatched, then that logic can be succinctly described in an alert monitor.

Also, when the values are gauges (can decrement) vs counters (never decrement) then it is more difficult to downsample data. As it stands, we can only know the current difference between inflight and dispatched and downsampling this data be lossy making it more difficult to see trends.

Reply:
Thanks @richardelling and glad to see your comments.
That makes sense.
I see that we may need three counters (which never decrements):
One at the place where dispatch is happening,
another at the place where thread started executing the IO
and, another at the place where thread completed executing the IO.